lib/internal/modules/esm/load.js

```diff
@@ -1,8 +1,66 @@
 'use strict';
 
+const {
+  ArrayPrototypePush,
+  RegExpPrototypeExec,
+  decodeURIComponent,
+} = primordials;
+
 const { defaultGetFormat } = require('internal/modules/esm/get_format');
-const { defaultGetSource } = require('internal/modules/esm/get_source');
 const { validateAssertions } = require('internal/modules/esm/assert');
+const { getOptionValue } = require('internal/options');
+const { fetchModule } = require('internal/modules/esm/fetch_module');
+
+// Do not eagerly grab .manifest, it may be in TDZ
+const policy = getOptionValue('--experimental-policy') ?
+  require('internal/process/policy') :
+  null;
+const experimentalNetworkImports =
+  getOptionValue('--experimental-network-imports');
+
+const { Buffer: { from: BufferFrom } } = require('buffer');
+
+const { readFile: readFileAsync } = require('internal/fs/promises').exports;
+const { URL } = require('internal/url');
+const {
+  ERR_INVALID_URL,
+  ERR_UNSUPPORTED_ESM_URL_SCHEME,
+} = require('internal/errors').codes;
+
+const DATA_URL_PATTERN = /^[^/]+\/[^,;]+(?:[^,]*?)(;base64)?,([\s\S]*)$/;
+
+async function getSource(url, context) {
```

Comment:
`getSource` is only 60 lines and while it used to be a separate hook, no longer is. In general I have a personal preference for fewer files since it's a better editing experience.